src/components/DateRangeInput/DateRangeInput.tsx

```diff
@@ -115,6 +125,7 @@ const DateRangeInput: React.FC<
   const [currentMonth, setCurrentMonth] = useState(currentDateRange[0] || now(timezone));
 
   const { isOpen, open, close } = useDisclosure();
+  const previousDateRange = usePrevious(datesToDayjs(value, timezone));
```

Comment:
Wouldn't that be more correct?

```suggestion
  const previousDateRange = usePrevious(currentDateRange);
```

Reply:
Nope, `currentDateRange` holds the input value, we need to check the validity against the controlled values and let the users decide the final outcome on "Cancel" or "Apply"